generator/src/lib.rs

```diff
@@ -115,13 +177,16 @@ fn parse_derive(ast: DeriveInput) -> (Ident, Generics, GrammarSource) {
         })
         .collect();
 
-    let argument = match grammar.len() {
-        0 => panic!("a grammar file needs to be provided with the #[grammar = \"PATH\"] or #[grammar_inline = \"GRAMMAR CONTENTS\"] attribute"),
-        1 => get_attribute(grammar[0]),
-        _ => panic!("only 1 grammar file can be provided"),
-    };
+    let mut grammar_sources = vec![];
```

Comment:
Maybe `Vec::with_capacity(grammar.len())`? I don't know how much this saves, especially if most folks will only be using one to two grammars, but seems worth it to me.